functionary/functionary/settings/django_.py

```diff
@@ -86,11 +86,24 @@
 # Database
 # https://docs.djangoproject.com/en/4.0/ref/settings/#databases
 
+DB_SQLITE = {
+    "ENGINE": "django.db.backends.sqlite3",
+    "NAME": BASE_DIR / "db.sqlite3",
+}
+
+DB_POSTGRESQL = {
+    "ENGINE": "django.db.backends.postgresql",
+    "NAME": f"{os.environ.get('DB_NAME', 'functionary')}",
+    "USER": f"{os.environ.get('DB_USER', 'admin')}",
+    "PASSWORD": f"{os.environ.get('DB_PASSWORD', 'password')}",
+    "HOST": f"{os.environ.get('DB_HOST', 'postgresql')}",
+    "PORT": f"{int(os.environ.get('DB_PORT', 5432))}",
+}
+
 DATABASES = {
-    "default": {
-        "ENGINE": "django.db.backends.sqlite3",
-        "NAME": BASE_DIR / "db.sqlite3",
-    }
+    "default": DB_POSTGRESQL
+    if os.environ.get("DB_ENGINE", "POSTGRESQL") == "POSTGRESQL"
```

Comment:
Implemented. I added logic to check if the database engine exists in the dictionary, if not, default to sqlite since that is the Django default. If DB_ENGINE is not defined, default to postgresql since that is our desired default. Let me know if you want me to change it so that postgresql is always the default, even if the user supplied an invalid database engine.

Reply:
I think the right thing long term is to not default to anything.  Think about the following situation:  Someone sets `DB_ENGINE` to "postgres" instead of "postgresql".  Instead of getting yelled at, the app carries on and just uses sqlite while the user may be blissfully unaware.  That could lead to some real frustration.

I was kind of suggesting that we could just punt on the error handling and have this blow up in ugly fashion for now (with the TODO comment reminding us to clean that up), but probably the only handling that would need to happen is to raise an [`ImproperlyConfigured`](https://docs.djangoproject.com/en/4.1/ref/exceptions/#improperlyconfigured) exception with some friendly message along the lines of:

> Invalid DB_ENGINE postgres. Valid choices are: ['sqlite', 'postgresql'].